crates/re_sdk/src/spawn.rs

```diff
@@ -156,34 +177,76 @@ pub fn spawn(opts: &SpawnOptions) -> Result<(), SpawnError> {
         return Ok(());
     }
 
-    let res = Command::new(executable_path)
+    fn check<T>(opts: &SpawnOptions, res: Result<T, std::io::Error>) -> Result<T, SpawnError> {
+        match res {
+            output @ Ok(_) => output.map_err(Into::into),
+            Err(err) if err.kind() == std::io::ErrorKind::NotFound => {
+                if let Some(executable_path) = opts.executable_path.as_ref() {
+                    Err(SpawnError::ExecutableNotFound {
+                        executable_path: executable_path.clone(),
+                    })
+                } else {
+                    Err(SpawnError::ExecutableNotFoundInPath {
+                        message: MSG_INSTALL_HOW_TO.to_owned(),
+                        executable_name: opts.executable_name.clone(),
+                        search_path: std::env::var("PATH").unwrap_or_else(|_| String::new()),
+                    })
+                }
+            }
+            Err(err) => Err(err.into()),
```

Comment:
This is a of a bit weird way of writing `map_err`

Reply:
:man_facepalming: